org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCodeActions.java

```diff
@@ -68,4 +68,23 @@ public List<CodeAction> doCodeActions(CodeActionContext context, Range range, DO
 		cancelChecker.checkCanceled();
 		return codeActions;
 	}
+
+	public CodeAction resolveCodeAction(CodeAction unresolved, DOMDocument document, SharedSettings sharedSettings,
+			CancelChecker cancelChecker) {
+		for (ICodeActionParticipant codeActionParticipant : extensionsRegistry.getCodeActionsParticipants()) {
+			try {
+				cancelChecker.checkCanceled();
+				if (codeActionParticipant.resolveCodeAction(unresolved, document, sharedSettings, extensionsRegistry,
+						cancelChecker)) {
+					return unresolved;

```

Comment:
it resolved but you're returning "unresolved". The naming sucks.

Reply:
I changed the API, this issue is fixed.